plugins/packageManager/packages.xql

```diff
@@ -122,13 +122,12 @@ declare %private function packages:public-repo-contents($installed as element(ap
         let $url := $config:REPO || "/public/apps.xml?version=" || packages:get-version() ||
             "&amp;source=" || util:system-property("product-source")
         (: EXPath client module does not work properly. No idea why. :)
-(:        let $request :=:)
-(:            <http:request method="get" href="{$url}" timeout="10">:)
-(:                <http:header name="Cache-Control" value="no-cache"/>:)
-(:            </http:request>:)
-(:        let $data := http:send-request($request):)
-        let $data := httpclient:get($url, false(), ())
-        let $status := xs:int($data/@statusCode)
+        let $request :=
+            <http:request method="get" href="{$url}" timeout="10">
+                <http:header name="Cache-Control" value="no-cache"/>
+            </http:request>
+        let $data := http:send-request($request)
+        let $status := xs:int($data[1]//http:response/@statusCode)
```

Comment:
The response status is returned in a `@status` attribute, not `@statusCode`. See http://expath.org/spec/http-client#d2e491.

Reply:
Also, since `http:send-request()` returns `<http:response>` as the first item, so `$data[1]/@status` should be sufficient as a replacement for `$data[1]//http:response/@status`.

Reply:
@joewiz Indeed. I have fixed that now. I couldn't test it until the public-xar-repo issue I mentioned above is resolved.

Reply:
Super, thanks so much.